### PR TITLE
added get and put apis

### PR DESCRIPTION
**Pull request by senilmaveric**

### CODE REVIEW

File: src/main/java/org/banker/loan/resources/LoanResource.java [hunk 1]
```diff
@@ -107,6 +110,36 @@ public Response  status(@PathParam("loanId") Long loanId, LoanStatus status, Uri
...
+    @Produces(MediaType.APPLICATION_JSON)
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Operation(summary = "search", description = "Retrieve a data by name,id,email and number")
+    @APIResponses(value = {
+            @APIResponse(responseCode = "200", description = "Retrieve of list of loan"),
+            @APIResponse(responseCode = "204", description = "No Content"),
+            @APIResponse(responseCode = "500", description = "Internal Server Error")
+    })
+    public ResponseDto search(
+            @QueryParam("searchValue") String searchValue, @QueryParam("page") int page, @QueryParam("size") int size) {
```
Comment (senthil-maveric) [inline src/main/java/org/banker/loan/resources/LoanResource.java:127]: Ensure input values are validated before processed. Are you handling the error response?

File: src/main/java/org/banker/loan/resources/LoanResource.java [hunk 2]
```diff
@@ -107,6 +110,36 @@ public Response  status(@PathParam("loanId") Long loanId, LoanStatus status, Uri
...
+    @Path("/{loanId}/{status}")
+    @Produces(MediaType.APPLICATION_JSON)
+    @Operation(summary = "updating the status", description = "Updating the status of the loans")
+    @APIResponses(value = {
+            @APIResponse(responseCode = "200", description = "updating loan status"),
+            @APIResponse(responseCode = "204", description = "No Content"),
+            @APIResponse(responseCode = "500", description = "Internal Server Error")
+    })
+    public ResponseDto statusUpdate(@PathParam("loanId") Long loanId,@PathParam("status") String status) {
+        String message=loanService.statusUpdate(loanId, status);
```
Comment (senthil-maveric) [inline src/main/java/org/banker/loan/resources/LoanResource.java:143]: Ensure input values are validated before processed. Are you handling the error response?

**Review by senthil-maveric — CHANGES_REQUESTED** [review]
Fix the comments

File: src/main/java/org/banker/loan/resources/LoanResource.java (hunk 1 above)
Comment (senilmaveric) [reply src/main/java/org/banker/loan/resources/LoanResource.java:127]: added validation and error response for fields

File: src/main/java/org/banker/loan/resources/LoanResource.java (hunk 2 above)
Comment (senilmaveric) [reply src/main/java/org/banker/loan/resources/LoanResource.java:143]: added validation and error response for fields

**Review by senthil-maveric — APPROVED** [review]

**Pull request merged**